### PR TITLE
[Francetvinfo.fr] Fix redirect loop

### PR DESCRIPTION
**Pull request by Bisaloo**
Tested on a fresh profile and everything looks fine.

### CODE REVIEW

**Comment by Bisaloo** [comment]
I'm so bad with regexes :(. If someone could help me, I would be very grateful.

**Comment by cschanaj** [comment]
@Bisaloo I assumed that you only want to rewrite the images on `^` and `www` and to exclude anything else.

```xml
<ruleset name="Franceinfo (partial)">
	<target host="francetvinfo.fr" />
	<target host="www.francetvinfo.fr" />

	<!-- do not test against ^ and www -->
	<exclusion pattern="^http://(www\.)?francetvinfo\.fr/$" />
	
		<test url="http://francetvinfo.fr/minify/1.2.309/fY9bDsIgEEU3JE-ttssZ6KDYAgpYYuLixcePCfVnJrnnJHNHUEm3fHhwZTQHZfZdD5r3-7HbbYUYpeykOuidARhULwfzYGmynpVSmE6JFVTJZiQp32ekNdn88lNw-B7_/GFFz0FPD0MG54JmQnDfobFVi5-sN451es70QWX_h1Fm_dgwX9LkFEUaM37XOHWbwsDQEB3HCbP2RmRAyRlJLkEXUNrJr6B_pu9b_VhGhxkfisaSX9gQ~.css" />
		<test url="http://francetvinfo.fr/skin/www/img/favicon/favicon.ico" />
		<test url="http://francetvinfo.fr/image/75eazbj8p-aea8/269/151/12994253.jpg" />
		<test url="http://francetvinfo.fr/skin/www/js/libs/xiti/4.5.8/xtcore.js" />
		<test url="http://francetvinfo.fr/skin/1.2.308/www/img/home/home-spritesmith.png" />
		<test url="http://francetvinfo.fr/skin/1.2.309/www/img/icn/apple/safari-pinned-tab.svg" />

		<test url="http://www.francetvinfo.fr/minify/1.2.309/fY9bDsIgEEU3JE-ttssZ6KDYAgpYYuLixcePCfVnJrnnJHNHUEm3fHhwZTQHZfZdD5r3-7HbbYUYpeykOuidARhULwfzYGmynpVSmE6JFVTJZiQp32ekNdn88lNw-B7_/GFFz0FPD0MG54JmQnDfobFVi5-sN451es70QWX_h1Fm_dgwX9LkFEUaM37XOHWbwsDQEB3HCbP2RmRAyRlJLkEXUNrJr6B_pu9b_VhGhxkfisaSX9gQ~.css" />
		<test url="http://www.francetvinfo.fr/skin/www/img/favicon/favicon.ico" />
		<test url="http://www.francetvinfo.fr/image/75eazbj8p-aea8/269/151/12994253.jpg" />
		<test url="http://www.francetvinfo.fr/skin/www/js/libs/xiti/4.5.8/xtcore.js" />
		<test url="http://www.francetvinfo.fr/skin/1.2.308/www/img/home/home-spritesmith.png" />
		<test url="http://www.francetvinfo.fr/skin/1.2.309/www/img/icn/apple/safari-pinned-tab.svg" />

	<rule from="^http://(www\.)?francetvinfo\.fr/(.+\.(?:css|ico|jpg|js|png|svg))$"
			to="https://www.francetvinfo.fr/$2" />
</ruleset>
```

File: src/chrome/content/rules/francetvinfo.fr.xml [hunk 1]
```diff
@@ -35,16 +30,39 @@ Fetch error: http://www.francetvinfo.fr/ => https://www.francetvinfo.fr/: Too ma
...
 
 -->
-<ruleset name="Franceinfo (partial)" platform="mixedcontent" default_off='failed ruleset test'>
+<ruleset name="Franceinfo (partial)">
 
 	<target host="francetvinfo.fr" />
 	<target host="www.francetvinfo.fr" />
+
+		<!-- Redirect to http -->
+		<exclusion pattern="http://(www\.)?francetvinfo.fr/([\w-/]*)(\.html)?$" />
```
Comment (cschanaj) [inline src/chrome/content/rules/francetvinfo.fr.xml]: Somehow the `rules-test` does not allow `*`, you must use `+` instead. You've forgotten to escape `.` and this `regex` match everything.

File: src/chrome/content/rules/francetvinfo.fr.xml [hunk 2]
```diff
@@ -35,16 +30,39 @@ Fetch error: http://www.francetvinfo.fr/ => https://www.francetvinfo.fr/: Too ma
...
 	<target host="mobile.francetvinfo.fr" />
+
+		<test url="http://mobile.francetvinfo.fr/" />
+		<test url="http://mobile.francetvinfo.fr/sciences/espace/video-trois-nouveaux-arrivants-a-bord-de-la-station-spatiale-internationale_2305263.amp" />
+	
 	<target host="stream.francetvinfo.fr" />
 
-	<rule from="^http://francetvinfo\.fr/"
-		to="https://www.francetvinfo.fr/" />
+	<rule from="^http://(www\.)?francetvinfo\.fr/(.*)\.(css|ico|jpg|js|png|svg)"
```
Comment (cschanaj) [inline src/chrome/content/rules/francetvinfo.fr.xml]: no `*` in `<rule from`

**Comment by Bisaloo** [comment]
@cschanaj, yeah, I just realized that and I got to my computer to test it. Thank you for your tips!

File: src/chrome/content/rules/francetvinfo.fr.xml [hunk 3]
```diff
@@ -35,16 +30,41 @@ Fetch error: http://www.francetvinfo.fr/ => https://www.francetvinfo.fr/: Too ma
...
 	<target host="mobile.francetvinfo.fr" />
+
+		<test url="http://mobile.francetvinfo.fr/" />
+		<test url="http://mobile.francetvinfo.fr/sciences/espace/video-trois-nouveaux-arrivants-a-bord-de-la-station-spatiale-internationale_2305263.amp" />
+	
 	<target host="stream.francetvinfo.fr" />
 
-	<rule from="^http://francetvinfo\.fr/"
-		to="https://www.francetvinfo.fr/" />
+	<rule from="^http://(www\.)?francetvinfo\.fr/(.+\.(?:css|ico|jpg|js|png|svg))"
```
Comment (J0WI) [inline src/chrome/content/rules/francetvinfo.fr.xml]: Please avoid non-capturing groups

Comment (J0WI) [inline src/chrome/content/rules/francetvinfo.fr.xml]: As far as I can see all these ressources are in paths like `/images`, `/skin`, `/script` or `/minify`. I think its safe to use this than `.+` with a list of file extensions. If you prefer file extensions, you should at a `$` on the end, to make sure that he regex only matches file extensions.

File: src/chrome/content/rules/francetvinfo.fr.xml [hunk 4]
```diff
@@ -35,16 +30,47 @@ Fetch error: http://www.francetvinfo.fr/ => https://www.francetvinfo.fr/: Too ma
...
+		<test url="http://www.francetvinfo.fr/minify/1.2.315/lZBhbsMgDIUvtEAZa5PdYXeYbOIlpIBTII0y9fDLuknRVPJjf-CBvyc_W4lnodXxhhpNa7A-NOpY6wboVbdQ6_ZDY6OMVnBqAfXp5SbT2QaJPFN8N-xHDhRykvgGn4tE/t57C2yCG9PRDzvMshySdxSSHy0RxkYYj_epKCyUO-44ptBTT3bDJ6n-4YBzIZJHIrRfHVPZygEhwtcmio7-vYr51eD-VvznIQHNylDPFPSKxseD2qiOPhZ4w5b6cvmPu/HFUQwC3ZmnRfchn1EM-Ubeg2VQZNDzEjQd7UA9gTrHuWnvLa-rpXNsE-Ttqz_074BQ~~.js" />
+		<test url="http://www.francetvinfo.fr/script/conf.js" />
+		<test url="http://www.francetvinfo.fr/script/conf.js?foo" />
+		<test url="http://www.francetvinfo.fr/skin/bower_components/webcomponentsjs/webcomponents-lite.js" />
+		<test url="http://www.francetvinfo.fr/skin/www/img/favicon/favicon.ico" />
+		<test url="http://www.francetvinfo.fr/skin/www/img/temp/dummy.gif" />
+		<test url="http://www.francetvinfo.fr/skin/1.2.309/www/img/icn/apple/57x57.png" />
+		<test url="http://www.francetvinfo.fr/skin/1.2.309/www/img/icn/apple/safari-pinned-tab.svg" />
+		<test url="http://www.francetvinfo.fr/skin/1.2.315/www/img/icn/ms/browserconfig.xml" /
```
Comment (J0WI) [inline src/chrome/content/rules/francetvinfo.fr.xml]: Missing closing bracket

Comment (Bisaloo) [inline src/chrome/content/rules/francetvinfo.fr.xml]: Thanks! I forgot to check the outcome of the tests.

**Pull request merged**